L1Trigger/L1TNtuples/plugins/L1UpgradeTfMuonTreeProducer.cc

```diff
@@ -117,18 +138,42 @@ L1UpgradeTfMuonTreeProducer::~L1UpgradeTfMuonTreeProducer() {}
 
 // ------------ method called to for each event  ------------
 void L1UpgradeTfMuonTreeProducer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::EDGetTokenT<l1t::RegionalMuonCandBxCollection> legacybmtfMuonToken;
+  edm::EDGetTokenT<l1t::RegionalMuonCandBxCollection> kbmtfMuonToken;
+  if (isEMU_) {
+    legacybmtfMuonToken = bmtfMuonToken_;
+    kbmtfMuonToken = bmtf2MuonToken_;
+  } else {
+    iEvent.getByToken(fedToken_, feds_);
+    // Get fw version
+    unsigned algoFwVersion { getAlgoFwVersion() };
+    if (algoFwVersion < 2499805536) {  //95000160(hex)
```

Comment:
lets do this with the Era switch in some of the future PRs

Reply:
The Era is unfortunately not fine grained enough in this case. The switch above also catches the handfull of of runs in 2018 that were taken with kBMTF triggering. If we used Eras here we would pollute the "BMTF tree" with events that should have gone in the "kBMTF tree" for 2018 data.